server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala

```diff
@@ -912,6 +914,245 @@ trait PushSubscriptionSetMethodContract {
+        |    ]
+        |  }""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "${SESSION_STATE.value}",
+           |    "methodResponses": [
+           |        [
+           |            "PushSubscription/set",
+           |            {
+           |                "notUpdated":{
+           |                    "${pushSubscription.id.serialise}":{
+           |                        "type":"invalidArguments",
+           |                        "description":"Expecting an array of JSON strings as an argument",
+           |                        "properties":["types"]
+           |                    }
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def updateShouldRejectBadType(server: GuiceJamesServer): Unit = {
+    val probe = server.getProbe(classOf[PushSubscriptionProbe])
+    val pushSubscription = probe
+      .createPushSubscription(username = BOB,
+        url = PushSubscriptionServerURL(new URL("https://example.com/push/?device=X8980fc&client=12c6d086")),
+        deviceId = DeviceClientId("12c6d086"),
+        types = Seq(MailboxTypeName))
+
+    val request: String =
+      s"""{
+        |    "using": ["urn:ietf:params:jmap:core"],
+        |    "methodCalls": [
+        |      [
+        |        "PushSubscription/set",
+        |        {
+        |            "update": {
+        |                "${pushSubscription.id.serialise}": {
+        |                  "types": ["Email", 36]
+        |                }
+        |              }
+        |        },
+        |        "c1"
+        |      ]
+        |    ]
+        |  }""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "${SESSION_STATE.value}",
+           |    "methodResponses": [
+           |        [
+           |            "PushSubscription/set",
+           |            {
+           |                "notUpdated":{
+           |                    "${pushSubscription.id.serialise}":{
+           |                        "type":"invalidArguments",
+           |                        "description":"Expecting an array of JSON strings as an argument",
+           |                        "properties":["types"]
+           |                    }
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
```

Comment:
We don't need to check `validated`? If it = `false` then updateType should be rejected?

Reply:
Where did you get that?

I just re-read the spec and cannot find anything related to this.

Reply:
When grooming, I remember

Reply:
Let me check the wrote on the issue

Reply:
I can't find it, maybe I was wrong, or we talk but did not write it down.
I just remember the key is "when a subscription is not validated, we need to restrict the update and limit properties when get method".